File: android-client/app/src/main/java/com/example/stripedemo/data/models/Account.kt
package com.example.stripedemo.data.models

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Account(
    val id: String,
    @Json(name = "stripe_account_id")
    val stripeAccountId: String,
    @Json(name = "stripe_customer_id")
    val stripeCustomerId: String?,
    val email: String?,
    @Json(name = "display_name")
    val displayName: String?,
    val created: String,
    @Json(name = "is_customer")
    val isCustomer: Boolean,
    @Json(name = "is_recipient")
    val isRecipient: Boolean,
    @Json(name = "is_onboarding")
    val isOnboarding: Boolean? = false,
    @Json(name = "customer_capabilities")
    val customerCapabilities: Map<String, Any?>? = null,
    @Json(name = "merchant_capabilities")
    val merchantCapabilities: Map<String, Any?>? = null
)

@JsonClass(generateAdapter = true)
data class AccountsResponse(
    val accounts: List<Account>
)

@JsonClass(generateAdapter = true)
data class CreateAccountRequest(
    val name: String,
    val email: String,
    val country: String = "US"
)

File: ios-client/StripeApplication/StripeApplication/Data/Models/Account.swift
import Foundation

struct Account: Codable, Identifiable, Hashable {
    let id: String
    let stripeAccountId: String
    let stripeCustomerId: String?
    let email: String?
    let displayName: String?
    let created: String
    let isCustomer: Bool
    let isRecipient: Bool
    let isOnboarding: Bool?
    let customerCapabilities: [String: AnyCodable]?
    let merchantCapabilities: [String: AnyCodable]?

    enum CodingKeys: String, CodingKey {
        case id
        case stripeAccountId = "stripe_account_id"
        case stripeCustomerId = "stripe_customer_id"
        case email
        case displayName = "display_name"
        case created
        case isCustomer = "is_customer"
        case isRecipient = "is_recipient"
        case isOnboarding = "is_onboarding"
        case customerCapabilities = "customer_capabilities"
        case merchantCapabilities = "merchant_capabilities"
    }

    static func == (lhs: Account, rhs: Account) -> Bool {
        lhs.id == rhs.id
    }

    func hash(into hasher: inout Hasher) {
        hasher.combine(id)
    }
}

struct AccountsResponse: Codable {
    let accounts: [Account]
}

struct CreateAccountRequest: Codable {
    let name: String
    let email: String
    let country: String

    init(name: String, email: String, country: String = "US") {
        self.name = name
        self.email = email
        self.country = country
    }
}
